docs/collisions.md

```diff
@@ -64,6 +64,61 @@ For example if a folder has five 5 with the same name say **_A.txt _** the five
 * [Extensions in folder names matter](https://apple.stackexchange.com/questions/123001/renamed-folder-becomes-a-file-with-an-extension) and should be handled :
 * Case sensitivity is taken into account. `Myfile` and `myFile` are different for example. That is useful to [ensure complete syncs](https://rclone.org/overview/#case-insensitive).
 
+#### Extensions
+
+For both files and folders the extension is any string that follows after the last dot.
+
+For example:
+<table>
+   <thead>
+      <th>File/Folder Name</th>
+      <th>Extension</th>
+      <th>Remark</th>
+   </thead>
+   <tbody>
+      <tr>
+         <td>My File.txt</td>
+         <td>txt</td>
+         <td>Regular</td>
+      </tr>
+      <tr>
+         <td>My File.txt(1)</td>
+         <td>txt(1)</td>
+         <td>Extension contains non-alphanumeric chars</td>
+      </tr>
+      <tr>
+         <td>My File.app</td>
+         <td>app</td>
+         <td>Regular</td>
+      </tr>
+      <tr>
+         <td>something.something</td>
+         <td>something</td>
+         <td>Name same as extension</td>
+      </tr>
+      <tr>
+         <td>2022.03.02</td>
+         <td>02</td>
+         <td>Extension is number</td>
+      </tr>
+      <tr>
+         <td>Folder.Work Files</td>
+         <td>Work Files</td>
+         <td>Extension contains spaces</td>
+      </tr>
+   </tbody>
+</table>
+
+As seen, extension can take infinite number of "shapes" BUT as long as all characters in the file name including the extension are valid file name chacters, then, **the extension is the string tha follows after the last dot** (Emphasis).
+
+
+#### Source of extensions
+
+a) The **source string for file extensions ON FILES** is the **`uploadFileName`** because that's the only place where it is maintained after a file is first uploaded in Permanent. In other words permanent takes a file and then removes its extension and then what is left is called the `displayName`
+
+b) The **source string for file extensions ON FOLDERS** is the `displayName` because Permanent does not (or at least did not ) do any kind of extension proccessing on folders, hence folder extensions are always visible in their `displayName`.
```

Comment:
Just curious: Do folders also have an `uploadName`, and we're just not using it here?  Or do they not have `uploadName` in the first place?

Reply:
Yes folders have `uploadName` however, it's not used because there's not a time we ever really need it. When a folder is first uploaded the `displayName === uploadName` unlike with files where the extension is ripped out of the `displayName`.

Reply:
When you say "it's not used" do you mean its value is NULL?  In other words, our proposed method of just always looking at `uploadFileName` wouldn't work, because for folders there's no string there at all, and we have to look at `displayName`?

If so, our conditional should not be `if (is_file(foo)): ...`, but rather `if (uploadName is not NULL): ... ; else ...look at displayName...`, I think?